### ISSUE TITLE
Source generation failure when using a `MethodDataSource` returning `IEnumerable<Func<string[]>>`

### ISSUE THREAD

**Issue by glen-84** [thomhurst/TUnit#1355]
```cs
public class Tests
{
    [Test]
    [MethodDataSource(nameof(TestData))]
    public async Task Test(string[] strings)
    {
        await Assert.That(strings).IsTypeOf<string[]>();
    }

    public static IEnumerable<Func<string[]>> TestData()
    {
        return
        [
            () => ["str1", "str2"],
            () => ["str3", "str4"]
        ];
    }
}
```

> CSC : error CS8785: Generator 'TestsGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.ArrayTypeSymbol' to type 'Microsoft.CodeAnalysis.INamedTypeSymbol'.'.

**Comment by thomhurst** [comment]
Thanks for the report! Should be fixed in v0.4.71

**Comment by glen-84** [comment]
Thanks a lot for the quick fix. 👍
